File: lession6/Chain Of Responsibility/java/Command/src/main/java/AddCommand.java
public class AddCommand implements Command {

    private OperationApi operation = null;

    private int opeNum;

    public void execute() {
        this.operation.add(opeNum);
    }

    public void undo() {
        this.operation.substract(opeNum);
    }

    public AddCommand(OperationApi operation, int opeNum) {
        this.operation = operation;
        this.opeNum = opeNum;
    }
}

File: lession6/Memento/csharp/Caculator/Caculator/AddCommand.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;


public class AddCommand : AbstractCommand
{
    private int opeNum;
    public AddCommand(OperationApi operation,int opeNum)
        :base(operation)
    {
        this.opeNum = opeNum;
    }
    public override void Execute()
    {
        this.operation.Add(this.opeNum);
    }
}

